core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -378,38 +355,31 @@ void OpenMP::impl_initialize(int thread_count) {
               << thread_count << " threads per process." << std::endl;
   }
   // Init the array for used for arbitrarily sized atomics
-  Impl::init_lock_array_host_space();
-}
+  init_lock_array_host_space();
 
-//----------------------------------------------------------------------------
+  initialized = true;
+}
 
-void OpenMP::impl_finalize() {
+void OpenMPInternal::finalize() {
   if (omp_in_parallel()) {
     std::string msg("Kokkos::OpenMP::finalize ERROR ");
-    if (!Impl::t_openmp_instance) msg.append(": not initialized");
+    if (this != &singleton()) msg.append(": not initialized");
     if (omp_in_parallel()) msg.append(": in parallel");
     Kokkos::Impl::throw_runtime_exception(msg);
   }
 
-  if (Impl::t_openmp_instance) {
+  if (this == &singleton()) {
+    auto instance = singleton();
```

Comment:
Consistency: in line 327 you have:
```
     auto &instance       = OpenMPInternal::singleton(); 
```
Also, in this case `instance` could maybe be `const`?

Reply:
done